abci/types/application.go

```diff
@@ -95,7 +95,20 @@ func (BaseApplication) ApplySnapshotChunk(req RequestApplySnapshotChunk) Respons
 }
 
 func (BaseApplication) PrepareProposal(req RequestPrepareProposal) ResponsePrepareProposal {
-	return ResponsePrepareProposal{ModifiedTxStatus: ResponsePrepareProposal_UNMODIFIED}
+	trs := make([]*TxRecord, len(req.Txs))
+	var total_bytes int64
+	for i, tx := range req.Txs {
+		if total_bytes > req.MaxTxBytes {
```

Comment:
Does this take into account the 0-to-default mapping?

Reply:
Sorry, I don't get what you mean by "0-to-default mapping"

Reply:
Did we not decide that requesting a maximum of zero would use the "default" maximum size? Or are we doing that mapping only at configuration time?

Reply:
Oh! I see now what you mean. That mapping will be part of the implementation we converged on in last ABCI++ meeting. For now, we're deferring it (possibly after v0.36... until some App developer insists they need it badly).
This PR is just to "prepare" both (a) the spec, and (b) the Apps present in the code, so that they don't break when the implementation is finally done.
So, `MaxTxBytes` above is really Tendermint telling the App "don't return me more than `MaxTxBytes` worth of transactions or I'll panic!".
Let me know if this is not clear.